Libraries/zephyr/MAX/CMakeLists.txt

```diff
@@ -0,0 +1,62 @@
+##############################################################################
+#
+# Copyright (C) 2023-2024 Analog Devices, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+##############################################################################
+
+
+if (${CONFIG_SOC} MATCHES "max32666")
+    set(TARGET_LC "max32665")
+elseif (${CONFIG_SOC} MATCHES "max326651")
+    set(TARGET_LC "max32650")
+else ()
+    set(TARGET_LC ${CONFIG_SOC})
+endif()
+string(TOUPPER ${TARGET_LC} TARGET_UC)
+
+zephyr_compile_definitions(
```

Comment:
@ozersa also suggest adding `-D${TARGET_UC}`

Reply:
Thanks, will consider on next update.

Reply:
@Jake-Carter I took a look it now, to apply it on https://github.com/analogdevicesinc/msdk/pull/996 
But a little confused.
Do you propose replacing 
    -DTARGET=${TARGET_UC} with -DTARGET=**-D**${TARGET_UC} ?

Reply:
@ozersa just to have both `TARGET` _and_ the actual value of `TARGET_UC` defined.

```CMake
zephyr_compile_definitions(
    -D${TARGET_UC}
    -DTARGET=${TARGET_UC}
    ...
)
```

Some places in the drivers need to check the target value.  Since the pre-processor can't do string comparisons sometimes `TARGET_UC` itself is checked.

```C
#ifdef MAX78000
// ...
#endif
```